src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -213,12 +224,13 @@ const AddCriteriaSelector: React.FC<AddCriteriaSelectorProps> = (props) => {
     options: [
       {
         label: 'Domains',
-        options: _.map((domainOption) => {
-          return {
-            value: domainOption,
+        options: _.map(
+          (domainOption) => ({
+            value: addKindToDomainOption(domainOption),
             label: domainOption.category,
-          };
-        }, datasetDetails.domainOptions),
+          }),
+          datasetDetails?.snapshotBuilderSettings?.domainOptions
```

Comment:
I don't think `datasetDetails` needs a `?.` here
```suggestion
          datasetDetails.snapshotBuilderSettings?.domainOptions
```